realm-jni/src/io_realm_internal_TableQuery.cpp

```diff
@@ -859,21 +861,270 @@ JNIEXPORT jlong JNICALL Java_io_realm_internal_TableQuery_nativeFind(
+          }
+
+          // import the handover query pointer using the background SharedGroup
+          std::unique_ptr<Query> query(SG(bgSharedGroupPtr)->import_from_handover(std::move(handoverQuery)));
+          if (isDifferentVersions) {
+              LangBindHelper::advance_read( *SG(bgSharedGroupPtr), *CH(replicationPtr) );
+          }
+
+          Table* table = query->get_table().get();
+
+          if (!QUERY_VALID(env, query.get()) || !ROW_INDEXES_VALID(env, table, start, end, limit))
+              return -1;
+
+            // run the query
+            TableView* tableView = new TableView( query->find_all(S(start), S(end), S(limit)) );
+
+            // sorting the results
+            if (!COL_INDEX_VALID(env, tableView, columnIndex))
+                      return -1;
+              int colType = tableView->get_column_type( S(columnIndex) );
+
+              switch (colType) {
+                  case type_Bool:
+                  case type_Int:
+                  case type_DateTime:
+                  case type_Float:
+                  case type_Double:
+                  case type_String:
+                      tableView->sort( S(columnIndex), ascending != 0 ? true : false);
+                      break;
+                  default:
+                      ThrowException(env, IllegalArgument, "Sort is currently only supported on integer, float, double, boolean, Date, and String columns.");
+                      return -1;
+              }
+
+            // handover the result
+            std::unique_ptr<SharedGroup::Handover<TableView> > handover = SG(bgSharedGroupPtr)->export_for_handover(*tableView, MutableSourcePayload::Move);
+
+            return reinterpret_cast<jlong>(handover.release());
+
+      } CATCH_STD()
+      return -1;
+  }
+
+JNIEXPORT jlong JNICALL Java_io_realm_internal_TableQuery_nativeFindAllMultiSortedWithHandover
+  (JNIEnv *env, jobject, jlong bgSharedGroupPtr, jlong replicationPtr, jlong queryPtr, jlong start, jlong end, jlong limit, jlongArray columnIndices, jbooleanArray ascending)
+  {
+
+      SharedGroup::Handover<Query>* handoverQueryPtr = reinterpret_cast<SharedGroup::Handover<Query>*>(queryPtr);
+      try {
+          jsize arr_len = env->GetArrayLength(columnIndices);
+          jsize asc_len = env->GetArrayLength(ascending);
+
+          jlong *long_arr = env->GetLongArrayElements(columnIndices, NULL);
+          jboolean *bool_arr = env->GetBooleanArrayElements(ascending, NULL);
+
+          if (arr_len == 0) {
+              ThrowException(env, IllegalArgument, "You must provide at least one field name.");
+              return -1;
+          }
+          if (asc_len == 0) {
+              ThrowException(env, IllegalArgument, "You must provide at least one sort order.");
+              return -1;
+          }
+          if (arr_len != asc_len) {
+              ThrowException(env, IllegalArgument, "Number of column indices and sort orders do not match.");
```

Comment:
I don't get your point it's the same message for `Java_io_realm_internal_TableView_nativeSortMulti` do you want to rename everywhere?